### ISSUE TITLE
Exception in plt.tight_layout()

### ISSUE THREAD

**Issue by voxspox** [matplotlib/matplotlib#8578]
I have an animated graph using `FuncAnimation`. I call tight_layout() on resize to fix the layout:

    fig.canvas.mpl_connect('resize_event', lambda *x: plt.tight_layout())

Mostly resizing the window works fine. Sometimes this leads to an exception:

    Traceback (most recent call last):
      File "/usr/lib/python3.5/tkinter/__init__.py", line 1553, in __call__
        return self.func(*args)
      File "/home/user/project/env/lib/python3.5/site-packages/matplotlib/backends/backend_tkagg.py", line 279, in resize
        self.resize_event()
      File "/home/user/project/env/lib/python3.5/site-packages/matplotlib/backend_bases.py", line 1833, in resize_event
        self.callbacks.process(s, event)
      File "/home/user/project/env/lib/python3.5/site-packages/matplotlib/cbook.py", line 554, in process
        proxy(*args, **kwargs)
      File "/home/user/project/env/lib/python3.5/site-packages/matplotlib/cbook.py", line 416, in __call__
        return mtd(*args, **kwargs)
      File "./monitor.py", line 199, in <lambda>
        fig.canvas.mpl_connect('resize_event', lambda *x: plt.tight_layout())
      File "/home/user/project/env/lib/python3.5/site-packages/matplotlib/pyplot.py", line 1406, in tight_layout
        fig.tight_layout(pad=pad, h_pad=h_pad, w_pad=w_pad, rect=rect)
      File "/home/user/project/env/lib/python3.5/site-packages/matplotlib/figure.py", line 1753, in tight_layout
        rect=rect)
      File "/home/user/project/env/lib/python3.5/site-packages/matplotlib/tight_layout.py", line 326, in get_tight_layout_figure
        max_nrows = max(nrows_list)
    ValueError: max() arg is an empty sequence
    Exception in Tkinter callback

I hope for someone this trace is useful. Unfortunately I can not provide a minimal example.

matplotlib==2.0.1

**Comment by tacaswell** [comment]
You should use `fig.tight_layout` rather that `plt.tight_layout`.  I suspect the problem is that you are sometimes getting the wrong figure from the state machine and then trying to run `tight_layout` on an empty figure.

This is reproducible with

```python
import matplotlib.pyplot as plt
fig = plt.gcf()
fig.tight_layout()
```

**Comment by anntzer** [comment]
`tight_layout()` on an empty figure should do nothing instead of raising an exception...

**Comment by tacaswell** [comment]
Originally thought that something like

```diff
@@ -323,8 +323,8 @@ def get_tight_layout_figure(fig, axes_list, subplotspec_list, renderer,
 
         subplots.append(ax)
 
-    max_nrows = max(nrows_list)
-    max_ncols = max(ncols_list)
+    max_nrows = max(nrows_list) if nrows_list else 0
+    max_ncols = max(ncols_list) if ncols_list else 0
 
     num1num2_list = []
     for subplotspec in subplotspec_list2:

```

would fix this, but that just moves the exception to elsewhere when there is an explicit check that numrows / numcols is 0.

**Comment by anntzer** [comment]
just do an early return in that case?

**Comment by tacaswell** [comment]
Agree with @anntzer this should just short-circuit and return a sensible no-op value (the only hard part will be sorting out _what_ that no-op value should be).

**Comment by voxspox** [comment]
hmm, I have only one figure. Should not make a difference.

But using fig is of course the clean way. I changed this in my code.

**Comment by tacaswell** [comment]
@voxspox You have one fig _now_ :smiling_imp:

**Comment by tacaswell** [comment]
Exact work to do there:

 - `lib/matplotlib/tight_layout.py:tight_layout_figure` return early if either nrows or ncols is 0
 - make sure the calling code (in `Figure.tight_layout`) also bails early
 - add a test

**Comment by tacaswell** [comment]
Fixed by #8891
